scripts/perf_scaling.py

```diff
@@ -196,45 +197,90 @@ def perf_file(location, filename, output_filename):
     lens = method_lengths_bytecode(filename)
 
     # Get all classes
-    classnames = {class_name(method) for method in lens.keys()}
-    lens = {method_name(method): len for method, len in lens.items()}
+    classname_to_methods = defaultdict(list)
+    for method in lens.keys():
+        class_name = get_class_name(method)
+        classname_to_methods[class_name].append(method)
+
+    lens = {get_method_name(method): len for method, len in lens.items()}
 
     csv = ""
-    for classname in classnames:
-        print(classname)
-
-        # TODO: hard-coded parameters
-        peggy_result = run_peggy(
-            classname,
-            location,
-            params,
-            optimization_level=optimization_level,
-            # TODO: adjust this timeout to be longer
-            timeout=1800,
-            container_name=config.docker_containername,
-        )
-        with open(output_filename, "a") as f:
-            match peggy_result:
-                case Result(ResultType.FAILURE, output):
-                    # TODO: write this to an output/log file instead
-                    with open(output_filename, "a") as f:
-                        f.write("Peggy failed")
-                case Result(ResultType.TIMEOUT, output):
-                    with open(output_filename, "a") as f:
-                        f.write("Peggy timed out")
-
-            # TODO: write this to an output/log file instead
-            f.write("PEGGY OUTPUT:")
-            f.write(peggy_result.output)
-            f.write("END PEGGY OUTPUT")
-
-        method_to_time = perf_from_output(str(peggy_result.output))
-
-        for method, times in method_to_time.items():
+    for classname, methods in classname_to_methods.items():
+        # Run one method at a time to get a more precise timeout
+        methods = [
+            # Constructor is <init>
+            method.replace(f" public {classname}(", " void <init>(")
+            .replace(f" ublic {classname}(", " void <init>(")
```

Comment:
0_0 what's ublic {classname} lol

Reply:
For some reason some of the constructors show up as `ublic`. It might be some issue with how I parse them or maybe with the bytes format. I'm not really sure. I'll leave a TODO to figure it out.

Reply:
That's okay, no worries